daemon/k8s_watcher.go

```diff
@@ -81,19 +97,76 @@ func k8sErrorHandler(e error) {
 	log.Error(e)
 }
 
-func (d *Daemon) createThirdPartyResources() error {
-	// TODO: Retry a couple of times
+func createCustomResourceDefinitions(clientset apiextensionsclient.Interface) error {
+	cnpCRDName := "ciliumnetworkpolicies." + k8s.CustomResourceDefinitionGroup
 
+	res := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cnpCRDName,
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   k8s.CustomResourceDefinitionGroup,
+			Version: k8s.CustomResourceDefinitionVersion,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:     "ciliumnetworkpolicies",
```

Comment:
Nit - would be better to have these strings as constants as some of them (e.g., the plural) are used elsewhere in the code.